app/api/puzzle/similarBatch/similarBatchFor.ts

```diff
@@ -13,20 +13,26 @@ import { getExistingUserRating } from '@/src/rating/getRating';
 import mongoose from 'mongoose';
 import similarity_distance from '@/src/similarity';
 
-// Returns empty array if no last batch is found.
-const similarBatchFor = async (user: User): Promise<Puzzle[]> => {
-  const lastBatch = await lastBatchFor(user);
-  if (lastBatch.length === 0) {
-    return lastBatch; // Exit early to avoid an unnecessary read.
-  }
+// We use a larger initial compromise (and so a larger initial radius)
+// as we'd like more similarity candidates to choose from, and this is
+// a batched mode, so more puzzle rating variance is acceptable.
+const INITIAL_COMPROMISE = 2;
+const MAX_COMPROMISE = 4;
```

Comment:
TODO: A bit of conflict here; in theory we want a large number of puzzle candidates, but we also don't want to take into account those outside the user's rating range. A robust similarity algorithm would not class puzzles of substantially differing difficulties as similar though.